server/src/main/java/io/crate/execution/dml/ArrayIndexer.java

```diff
@@ -36,10 +39,19 @@
 
 public class ArrayIndexer<T> implements ValueIndexer<List<T>> {
 
+    public static String toArrayLengthFieldName(ColumnIdent arrayColumnIdent) {
+        return ARRAY_LENGTH_FIELD_PREFIX + arrayColumnIdent.name();
+    }
```

Comment:
I think we should use oid instead. We can drop column and re-add another column ((even with another type) but with previously existed name. 

Actually, not oid but `Reference.storageIdent` which is oid for >=5.5 tables and name for 5.4

Reply:
Thank you, a quick question, let say for `5.9` if we create an array with oid `1`, the array index field should be `_array_length_1` or of its own oid in `long`?

Reply:
ARRAY_LENGTH_FIELD_PREFIX + storageIdent ==> _array_length_1

Reply:
Oh sorry, it should be `ARRAY_LENGTH_FIELD_PREFIX` + `ref.storageIdentLeafName`

Reply:
storageIdent/storageIdentLeafName is the same for >=5.5, it's OID anyway but it matters for <5.5 tables.

Reply:
Thanks, because the array field name includes the oid of the array, dropping and re-adding the array works flawlessly.